kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiTHttpFrontendService.scala

```diff
@@ -0,0 +1,285 @@
+      httpConf.setRequestHeaderSize(requestHeaderSize)
+      httpConf.setResponseHeaderSize(responseHeaderSize)
+      val connectionFactory = new HttpConnectionFactory(httpConf)
+
+      val useSsl = conf.get(FRONTEND_THRIFT_HTTP_USE_SSL)
+      val schemeName = if (useSsl) "https" else "http"
+
+      // Change connector if SSL is used
+      val connector =
+        if (useSsl) {
+          val keyStorePath = conf.get(FRONTEND_THRIFT_HTTP_SSL_KEYSTORE_PATH)
+
+          if (keyStorePath.isEmpty) {
+            throw new IllegalArgumentException(FRONTEND_THRIFT_HTTP_SSL_KEYSTORE_PATH.key +
+              " Not configured for SSL connection")
+          }
+
+          val keyStorePassword = conf.get(FRONTEND_THRIFT_HTTP_SSL_KEYSTORE_PASSWORD)
+          if (keyStorePassword.isEmpty) {
+            throw new IllegalArgumentException(FRONTEND_THRIFT_HTTP_SSL_KEYSTORE_PASSWORD.key +
+              " Not configured for SSL connection")
+          }
+
+          val sslContextFactory = new SslContextFactory.Server
+          val excludedProtocols = conf.get(FRONTEND_THRIFT_HTTP_SSL_PROTOCOL_BLACKLIST).split(",")
+          info("HTTP Server SSL: adding excluded protocols: " +
+            String.join(",", excludedProtocols: _*))
+          sslContextFactory.addExcludeProtocols(excludedProtocols: _*)
+          info("HTTP Server SSL: SslContextFactory.getExcludeProtocols = " +
+            String.join(",", sslContextFactory.getExcludeProtocols: _*))
+          sslContextFactory.setKeyStorePath(keyStorePath.get)
+          sslContextFactory.setKeyStorePassword(keyStorePassword.get)
+          new ServerConnector(
+            server.get,
+            sslContextFactory,
+            connectionFactory)
+        } else {
+          new ServerConnector(server.get, connectionFactory)
+        }
+
+      connector.setPort(portNum)
+      // Linux:yes, Windows:no
+      connector.setReuseAddress(!OSUtils.IS_WINDOWS)
+      val maxIdleTime = conf.get(FRONTEND_THRIFT_HTTP_MAX_IDLE_TIME)
+      connector.setIdleTimeout(maxIdleTime)
+      connector.setAcceptQueueSize(maxThreads)
+      server.foreach(_.addConnector(connector))
+
+      val processor = new TCLIService.Processor[TCLIService.Iface](this)
+      val protocolFactory = new TBinaryProtocol.Factory
+      val servlet = new ThriftHttpServlet(processor, protocolFactory, authFactory, conf)
+      servlet.init()
+
+      // Context handler
+      val context = new ServletContextHandler(ServletContextHandler.SESSIONS)
+      context.setContextPath("/")
+
+      context.addEventListener(new ServletContextListener() {
+        override def contextInitialized(servletContextEvent: ServletContextEvent): Unit = {
+          val metrics = MetricsFactory.getInstance
+          if (metrics != null) {
+            try {
+              metrics.incrementCounter(MetricsConstant.OPEN_CONNECTIONS)
+              metrics.incrementCounter(MetricsConstant.CUMULATIVE_CONNECTION_COUNT)
```

Comment:
Glad to see we report some metrics here. Can we also make sure if it is suitable for Kyuubi's metric system?